urlfb310a588a_alokhan_LinkStateNetworkSimulation_tgz-pJ8-reso_examples_static_routing_DemoJ8/src/dijkstra/FibonacciHeap.java

```diff
@@ -0,0 +1,584 @@
+     */
+    public static FibonacciHeap union(
+            FibonacciHeap h1,
+            FibonacciHeap h2) {
+        FibonacciHeap h = new FibonacciHeap();
+
+        if ((h1 != null) && (h2 != null)) {
+            h.minNode = h1.minNode;
+
+            if (h.minNode != null) {
+                if (h2.minNode != null) {
+                    h.minNode.right.left = h2.minNode.left;
+                    h2.minNode.left.right = h.minNode.right;
+                    h.minNode.right = h2.minNode;
+                    h2.minNode.left = h.minNode;
+
+                    if (h2.minNode.key < h1.minNode.key) {
+                        h.minNode = h2.minNode;
+                    }
+                }
+            } else {
+                h.minNode = h2.minNode;
+            }
+
+            h.nNodes = h1.nNodes + h2.nNodes;
+        }
+
+        return h;
+    }
+
+    // union
+    /**
+     * Creates a String representation of this Fibonacci heap.
+     *
+     * @return String of this.
+     */
+    public String toString() {
+        if (minNode == null) {
+            return "FibonacciHeap=[]";
+        }
+
+        // create a new stack and put root on it
+        Stack<FibonacciHeapNode> stack = new Stack<FibonacciHeapNode>();
+        stack.push(minNode);
+
+        StringBuffer buf = new StringBuffer(512);
+        buf.append("FibonacciHeap=[");
+
+        // do a simple breadth-first traversal on the tree
+        while (!stack.empty()) {
+            FibonacciHeapNode curr = stack.pop();
+            buf.append(curr);
+            buf.append(", ");
+
+            if (curr.child != null) {
+                stack.push(curr.child);
+            }
+
+            FibonacciHeapNode start = curr;
+            curr = curr.right;
+
+            while (curr != start) {
+                buf.append(curr);
+                buf.append(", ");
+
+                if (curr.child != null) {
+                    stack.push(curr.child);
+                }
+
+                curr = curr.right;
+            }
+        }
+
+        buf.append(']');
+
+        return buf.toString();
+    }
+
+    // toString
+    /**
+     * Performs a cascading cut operation. This cuts y from its parent and then
+     * does the same for its parent, and so on up the tree.
+     *
+     * <p>
+     * Running time: O(log n); O(1) excluding the recursion</p>
+     *
+     * @param y node to perform cascading cut on
+     */
+    protected void cascadingCut(FibonacciHeapNode y) {
+        FibonacciHeapNode z = y.parent;
+
+        // if there's a parent...
+        if (z != null) {
+            // if y is unmarked, set it marked
+            if (!y.mark) {
+                y.mark = true;
+            } else {
+                // it's marked, cut it from parent
+                cut(y, z);
+
+                // cut its parent as well
+                cascadingCut(z);
+            }
+        }
+    }
+
+    // cascadingCut
+    protected void consolidate() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 475-502 into a separate method.